src/v2/view-builder/utils/AuthenticatorUtil.js

```diff
@@ -181,12 +182,16 @@ const getAuthenticatorData = function(authenticator, isVerifyAuthenticator) {
 
   case AUTHENTICATOR_KEY.CUSTOM_APP: {
     Object.assign(authenticatorData, {
-      description: '',
-      iconClassName: 'mfa-custom-app',
+      description: '', // TODO: Yet to decide on the description OKTA-480733
       buttonDataSeAttr: getButtonDataSeAttr(authenticator),
     });
     break;
   }
+
+  }
+  // Add logo URI, Logo URI must take precedence than icon class name
+  if(authenticator?.relatesTo?.logoUri) {
+    authenticatorData.logoUri = authenticator.relatesTo.logoUri;
```

Comment:
if `logoUri` is specific to `custom_app` can this line be moved within that case check `case AUTHENTICATOR_KEY.CUSTOM_APP`

Reply:
I do not want to restrict it only for `custom_app`. in future, if any auth have `logoUri` prop then it can take precedence.

Reply:
nit: there are 15 auth keys and this if condition would be hit for all of those even though its relevant just to custom_app. I find it unnecessary unless there is any plan to add logo uri for other authenticators in near future.

Reply:
moved inside switch.